### PR TITLE
Update Doctrine ORM and DBAL

### PR DESCRIPTION
**Pull request by Sperling-0**
- Use AddressType enum

Ticket: https://phabricator.wikimedia.org/T356744

### CODE REVIEW

File: composer.json
Comment (moiikana) [inline composer.json:24]: needs new release versions as soon as those branches are merged

File: app/Controllers/API/Donation/AddressChangeController.php
Comment (Abban) [inline app/Controllers/API/Donation/AddressChangeController.php:14]: You can delete the `ADDRESS_TYPE_COMPANY` and `MESSAGE_NO_IDENTIFIER` consts from this class

**Review by Abban — APPROVED** [review]

**Comment by moiikana** [comment]
[ Do not merge until Memory Leak investigation is done ]

**Pull request merged**
